.github/workflows/build-msvc.yml

```diff
@@ -71,19 +71,49 @@
             C:\cygwin-packages
           key: cygwin-packages
 
+      - name: Get Autoconf cache key
+        shell: bash
+        env:
+          HOST: ${{ matrix.x86_64 && 'x86_64-pc-windows' || 'i686-pc-windows' }}
+          CC: ${{ matrix.cc }}
+        run: >-
+          echo "AUTOCONF_CACHE_KEY=$HOST-$CC-$({ cat configure; uname; } | sha1sum | cut -c 1-7)" >> $GITHUB_ENV
+
+      - name: Restore Autoconf cache
+        uses: actions/cache/restore@v4
+        env:
+          AUTOCONF_CACHE_KEY: ${{ env.AUTOCONF_CACHE_KEY }}
+        with:
+          path: |
+            ocaml-default.cache
+          key: ${{ env.AUTOCONF_CACHE_KEY }}
+
       - name: Build OCaml
-        shell: bash -e {0}
+        shell: bash
         env:
           HOST: ${{ matrix.x86_64 && 'x86_64-pc-windows' || 'i686-pc-windows' }}
           CC: ${{ matrix.cc }}
+          AUTOCONF_CACHE_KEY: ${{ env.AUTOCONF_CACHE_KEY }}
         run: >-
           eval $(tools/msvs-promote-path) ;
-          ./configure --host=$HOST CC=$CC;
+          if ! ./configure --cache-file=ocaml-default.cache --host=$HOST CC=$CC ; then
```

Comment:
Why not use the standard `config.cache` name, and `-C` short option?

Reply:
I'd rather have the cache file name explicit because of the `git config ocaml.configure-cache` setting; and the default name when a host is specified is `ocaml-$host.cache`. It's even also written in the parent directory.